### PR TITLE
update horizons map gt

### PR DESCRIPTION
**Pull request by gkarnati3**

### CODE REVIEW

File: public/css/style.css [hunk 1]
```diff
@@ -0,0 +1,177 @@
+@font-face {
```
Comment (mjkaufer) [inline public/css/style.css]: Why do we have both a `main.css` and `style.css`? Is one file meant for a certain type of styling? If so, we should mention that and rename the files to be more specific so future contributors know how to organize the CSS

Comment (mjkaufer) [inline public/css/style.css]: Why do we have both a main.css and style.css? Is one file meant for a certain type of styling? If so, we should mention that and rename the files to be more specific so future contributors know how to organize the CSS

File: public/css/style.css [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+}
+a:hover {
+  border-bottom-style: solid;
+}
+
+::-webkit-input-placeholder { /* WebKit, Blink, Edge */
+  color: var(--text-color);
+}
+
+#email-rendered {
```
Comment (mjkaufer) [inline public/css/style.css]: I don't know why we have these styles and the form styles – can we get rid of useless CSS styles? I'm sure there's an auditor somewhere that'll tell you which styles a page is using

File: public/css/toggle.css [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+.btn:hover {
+  box-shadow: inset 0 -3px 0 rgba(0, 0, 0, 0.5);
+}
+
+.btn:focus {
+  outline: none;
+  background-color: #f6a786;
+}
+
+/* .card {
```
Comment (mjkaufer) [inline public/css/toggle.css]: Delete commented code

File: public/css/main.css [hunk 4]
```diff
@@ -0,0 +1,194 @@
+.standalone {
+    display: block;
+}
+
+.left-panel {
+    width: 74%;
+    float: left;
```
Comment (mjkaufer) [inline public/css/main.css]: Let's not use floats if we can help it – try playing with CSS grid – [this](https://cssgridgarden.com/) is a really good tutorial I found

File: public/css/main.css [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    /* manually found this breakpoint by trial/error */
+    /* make floor map centered */
+    .maps {
+        width: 90%;
+        left: 53%;
+    }
+}
+
+.alignleft {
+	float: left;
```
Comment (mjkaufer) [inline public/css/main.css]: Fix tabbing, also see earlier comment about trying to avoid floats

File: public/event-data.json [hunk 6]
```diff
@@ -0,0 +1,128 @@
+[
```
Comment (mjkaufer) [inline public/event-data.json:1]: We don't need to push this file as it's from BuildGT and is out of date – instead, can we make a file called `event-data.template.json` with two generic sample events so people adding the schedule in the future know what to write to the JSON

File: public/index.html [hunk 7]
```diff
@@ -1,22 +1,60 @@
...
-            <input type="checkbox" onclick="toggleFloors()">
-            <span class="slider round"></span>
-        </label>
 
-        <object id="floor1" type="image/svg+xml" data="assets/maps/ICF1Edit.svg"></object>
-        <object id="floor2" type="image/svg+xml" data="assets/maps/ICF2Edit.svg"></object>
+<head>
+    <meta charset="utf-8">
+    <title>MapGT</title>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
```
Comment (mjkaufer) [inline public/index.html:7]: Can we get rid of jQuery? We're only using it for `getJSON` which is super easy to do with vanilla JS

File: public/js/util.js [hunk 8]
```diff
@@ -0,0 +1,247 @@
+var firstFloor = document.getElementById("floor1");
+var secondFloor = document.getElementById("floor2");
+var modal = document.querySelector('.modal');
+var modalOverlay = document.querySelector('.modal-overlay');
+
+var events;
+
+// An AJAX request to get the data from the event-data.json file and parsing it
+$.getJSON({
```
Comment (mjkaufer) [inline public/js/util.js:9]: no jquery

File: public/js/util.js [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+    url: "./event-data.json",
+    async: false
+}, function(data) {
+    events = data;
+});
+
+// Transforming the start and end times into date objects
+for (event of events) {
```
Comment (mjkaufer) [inline public/js/util.js]: This is going to have a race condition that is definitely gonna fail – should put in callback after getting JSON

File: public/js/util.js [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+    var eventStartTime = event.startTime;
+    var eventEndTime = event.endTime;
+    event.startTime = new Date("March 2, 2019" + eventStartTime);
+    event.endTime = new Date("March 2, 2019 " + eventEndTime);
+}
+
+console.log(events);
```
Comment (mjkaufer) [inline public/js/util.js]: no console logs when merging to master

File: public/js/util.js [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+
+var rooms = [
```
Comment (mjkaufer) [inline public/js/util.js:28]: Delete this since we have the JSON

File: public/js/util.js [hunk 12]
```diff
@@ -0,0 +1,247 @@
...
+    var stringRoomNumber = String(roomNumber);
+    var floorNumber = parseInt(stringRoomNumber.charAt(0));
+    var floorElement = floorNumber == 2 ? secondFloor : firstFloor;
+    floorElement.addEventListener("load", () => {
+        var svgDOM = floorElement.contentDocument.children[0]
+        var matchingTextNodes = Array.from(svgDOM.querySelectorAll('text'))
+            .find(el => el.textContent.trim() === stringRoomNumber.trim());
+        // x offset +4
+        // y offset -25
+        var xpos = matchingTextNodes.getAttribute("x") - -4;
```
Comment (mjkaufer) [inline public/js/util.js:97]: Can we just say +4 here lol

File: public/js/util.js [hunk 13]
```diff
@@ -0,0 +1,247 @@
...
+
+function createPin(roomNumber, location) {
+    // creates a pin element that will be added to the map svg
+
+    var pin = document.createElement("object");
+    setPinAttributes(pin, roomNumber, location);
+    return pin
+}
+
+function tog1() {
```
Comment (mjkaufer) [inline public/js/util.js:134]: There shouldn't be a `tog1` and `tog2`, ideally we have a `highlightFloor(floorNumber)` method

File: public/css/toggle.css [hunk 14]
```diff
@@ -0,0 +1,183 @@
...
+  font-family: 'Open Sans', sans-serif;
+  font-size: 14px;
+  background-color: #42626d;
+}
+
+.btn:hover {
+  box-shadow: inset 0 -3px 0 rgba(0, 0, 0, 0.5);
+}
+
+.btn:focus {
```
Comment (mjkaufer) [inline public/css/toggle.css]: You shouldn't use `focus` to give the buttons the highlight effect because
* The floor 1 button isn't highlighted by default
* When you click something other than the button, the button gets unhighlighted

Instead, there should be a selector for `.btn.selected` or something, and clicking a button adds the `.selected` class to that button while removing it from other buttons

File: public/js/util.js [hunk 15]
```diff
@@ -0,0 +1,247 @@
...
+window.onclick = function() {
+    removeModal();
+}
+
+function dispModal() {
+    toggleSelectedPin();
+    toggleModal();
+}
+
+function toggleSelectedPin() {
```
Comment (mjkaufer) [inline public/js/util.js:228]: What's this for?

File: public/style.css [hunk 16]
```diff
@@ -65,4 +65,38 @@ input:checked + .slider:before {
 #floor2 {
```
Comment (mjkaufer) [inline public/style.css]: Why is there a CSS file in the root of the `public` folder, along with CSS files in `public/css`

**Review by mjkaufer — CHANGES_REQUESTED** [review]

File: public/css/main.css [hunk 17]
```diff
@@ -0,0 +1,194 @@
...
+    position: absolute;
+    width: 50%;
+    /* top: 50%; */
+    bottom: 15%;
+    left: 48%;
+    height: 80%;
+    transform: translateX(-75%);
+}
+
+@media (max-width:1200px) {
```
Comment (evan10s) [inline public/css/main.css]: In same vein as Matt's comment about this file, use a CSS grid and you won't need this trial and error stuff, so it'll be more flexible across devices

File: public/css/toggle.css [hunk 18]
```diff
@@ -0,0 +1,183 @@
...
+
+.toggleButton {
+  bottom: 5%;
+  left: 52%;
+  transform: translateX(-140%);
+  position: absolute;
+}
+
+
+@media (max-width:1300px) {
```
Comment (evan10s) [inline public/css/toggle.css]: CSS grids can probably remove need for all these media queries as well.  Don't repeat yourself; if you have to copy and paste repeatedly there's probably a better/easier way to do whatever you're trying to do

File: public/js/util.js [hunk 19]
```diff
@@ -0,0 +1,247 @@
...
+    var floorElement = floorNumber == 2 ? secondFloor : firstFloor;
+    floorElement.addEventListener("load", () => {
+        var svgDOM = floorElement.contentDocument.children[0]
+        var matchingTextNodes = Array.from(svgDOM.querySelectorAll('text'))
+            .find(el => el.textContent.trim() === stringRoomNumber.trim());
+        // x offset +4
+        // y offset -25
+        var xpos = matchingTextNodes.getAttribute("x") - -4;
+        // I know this looks weird, but javascript is more weird
+        // if i do xpos + 4, it just doesnt work, but - works
```
Comment (evan10s) [inline public/js/util.js:99]: This is because `getAttribute` is returning a string.  When you use `+` and one of the operands is a string (`"4" + 5`), you'll get the concatenation of the strings back (`45`).  The reason using `-` works is because you can't subtract strings, so JavaScript implicitly converts the string operand to a number.  You can try this in your browser's JS console and see for yourself, too.

File: public/js/util.js [hunk 20]
```diff
@@ -0,0 +1,247 @@
...
+        var ypos = matchingTextNodes.getAttribute("y") - 40;
```
Comment (evan10s) [inline public/js/util.js:100]: You should parse the `getAttribute` call to make it explicitly a number

File: public/js/util.js [hunk 21]
```diff
@@ -0,0 +1,247 @@
...
+        var pins = document.getElementById("pins");
+        pins.appendChild(pin);
+        pin.addEventListener("load", () => {
+            pinDOM = pin.contentDocument.children[0];
+            pinDOM.setAttribute("x", xpos);
+            pinDOM.setAttribute("y", ypos);
+            svgDOM.appendChild(pinDOM);
+            pinDOM.id = "room-" + roomNumber
+            pinDOM.addEventListener("click", function(e) {
+                var clickedElement = getTopLevelOfPin(e.target)
```
Comment (evan10s) [inline public/js/util.js]: missing semicolon (please fix the other missing semicolons in the rest of this file, too)

File: public/js/util.js [hunk 22]
```diff
@@ -0,0 +1,247 @@
...
+    addLocationPins(1);
+}
+
+function tog2() {
+    firstFloor.style.display = "none";
+    secondFloor.style.display = "block";
+    addLocationPins(2);
+}
+
+// $('.toggle').click(function(e) {
```
Comment (evan10s) [inline public/js/util.js]: remove comment if unused

**Review by evan10s — CHANGES_REQUESTED** [review]

File: public/js/util.js (hunk 19 above)
Comment (Litagano-M) [reply public/js/util.js:99]: #JustJavascriptThings

File: public/index.html [hunk 23]
```diff
@@ -35,19 +35,24 @@ <h1 class="alignright">MapGT</h1>
...
+            <div class="toppanel">
+                <div class="event-card">
+                    <h2>Now</h2>
+                    <p id="current-event"></p>
+                </div>
+            </div>
+            <div class="bottompanel">
+            <div class="more-info">
+                <h2>Upcoming</h2>
+                <div id="upcoming-events" style="white-space: pre-line"></div>
```
Comment (mjkaufer) [inline public/index.html]: We shouldn't have any in-line style – add `white-space: pre-line` to CSS for `.upcoming-events`

File: public/css/main.css [hunk 24]
```diff
@@ -12,6 +12,21 @@
     height: 725px;
     float: right;
     width: 25%;
+    border: solid; 
+}
+
+.toppanel {
+    width: 100%;
+    height: 330px;
```
Comment (mjkaufer) [inline public/css/main.css]: Can we make `.toppanel` and `.bottompanel` responsive – if you're on a small screen, you won't be able to see these panels. Look into `vh` – something like `40vh` would be a good thought

File: public/css/toggle.css [hunk 25]
```diff
@@ -53,15 +53,15 @@
 }
 
 .event-card {
-  transform: translate(110%, 30%);
+  transform: translate(4%, 15%);
```
Comment (mjkaufer) [inline public/css/toggle.css]: Don't use `translate` for `event-card` and `more-info` – the way you're using it, this looks like a job for `margin` or `padding`

File: public/index.html [hunk 26]
```diff
@@ -25,7 +25,7 @@ <h1 class="alignright">MapGT</h1>
         </div>
     
         <div class="section toggleButton">
-            <button class="btn btn--floor1" onclick="tog1()">1st</button>
+            <button class="btn btn--floor1 active" id="removeactive" onclick="tog1()">1st</button>
```
Comment (mjkaufer) [inline public/index.html:22]: I don't like that one of the buttons has an id while the other doesn't – it's kinda weird. Why don't we just get elements by class name `btn`, remove the `active` class from all of them, and add the class to the button that was just clicked? I think it'd be a lot more elegant and extensible in the future if we add more buttons

**Review by mjkaufer — CHANGES_REQUESTED** [review]

File: public/js/util.js [hunk 27]
```diff
@@ -0,0 +1,229 @@
...
+    async: false
+}, function(data) {
+    events = data;
+});
+
+// Transforming the start and end times into date objects
+for (event of events) {
+    var eventStartTime = event.startTime;
+    var eventEndTime = event.endTime;
+    event.startTime = new Date("March 2, 2019" + eventStartTime);
```
Comment (mjkaufer) [inline public/js/util.js]: March 2nd shouldn't be hardcoded – at the very least, make a constant in the JS file, but the better option is to encode it into the JSON file since this will break for multi-day events (like horizons)

**Review by mjkaufer — CHANGES_REQUESTED** [review]
